src/test/java/picard/sam/markduplicates/MarkDuplicatesIntegrationTests.java

```diff
@@ -0,0 +1,378 @@
+package picard.sam.markduplicates;
+
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMFlag;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.util.IOUtil;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import picard.cmdline.CommandLineProgramTest;
+import picard.sam.DuplicationMetrics;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class MarkDuplicatesIntegrationTests extends CommandLineProgramTest {
+
+    protected static String TEST_BASE_NAME = null;
+    protected static File TEST_DATA_DIR = null;
+
+    @BeforeClass
+    public void setUp() {
+        TEST_BASE_NAME = "MD_IT";
+        TEST_DATA_DIR = new File("testdata/picard/sam/MarkDuplicates/IntTest");
+    }
+
+    @DataProvider(name = "MarkDuplicatesIntegrationTest")
+    public Object[][] createMarkDuplicatesIntegrationTest() {
+        return new Object[][]{
+                {"duplicates_test.sam"},
+                {"duplicates_big_test.sam"},
+                {"optical_duplicates_test.sam"},
+        };
+    }
+
+    @DataProvider(name = "SecondarySupplementaryUnmappedTest")
+    public Object[][] createSecondarySupplementaryUnmappedTest() {
+        return new Object[][]{
+                {"secondary_supplementary_test.sam"},
+                {"secondary_supplementary_small_test.sam"},
+                {"secondary_supplementary_unmapped_big_test.bam"}
+        };
+    }
+
+    @Test(dataProvider = "MarkDuplicatesIntegrationTest")
+    public void markDuplicatesWithoutRemovingDuplicatesIntegrationTest(final String input) {
+        final File outputDir = IOUtil.createTempDir(TEST_BASE_NAME + ".", ".tmp");
+        outputDir.deleteOnExit();
+        final ArrayList<String> args = new ArrayList<>();
+        args.add("INPUT=" + new File(TEST_DATA_DIR,input).getAbsolutePath());
+        final File output = new File(outputDir, TEST_BASE_NAME + ".sam");
+        args.add("OUTPUT=" + output.getAbsolutePath());
+        final File metrics = new File(outputDir, TEST_BASE_NAME + ".integration_metrics");
+        args.add("METRICS_FILE=" + metrics.getAbsolutePath());
+
+        Assert.assertEquals(runPicardCommandLine(args), 0);
+        examineDuplicationMetrics(input, output, metrics, false);
+    }
+
+    @Test(dataProvider = "MarkDuplicatesIntegrationTest")
+    public void markDuplicatesWithRemovingDuplicatesIntegrationTest(final String input) {
```

Comment:
`markDuplicatesWithRemovingDuplicatesIntegrationTest` and `secondarySupplementaryUnmappedWithRemovingDuplicatesIntegrationTest` are exactly the same. The contents of the test should be pulled out into its own test and the two separate data providers should be one line methods that call the abstracted test.